python/pyspark/sql/dataframe.py

```diff
@@ -1223,10 +1226,8 @@ def withColumnRenamed(self, existing, new):
         >>> df.withColumnRenamed('age', 'age2').collect()
         [Row(age2=2, name=u'Alice'), Row(age2=5, name=u'Bob')]
         """
-        cols = [Column(_to_java_column(c)).alias(new)
-                if c == existing else c
-                for c in self.columns]
-        return self.select(*cols)
+        assert existing in self.columns, "%s is not an existing column" % existing
```

Comment:
This is not consistent with scala version, see the [doc](https://github.com/apache/spark/blob/master/sql/core/src/main/scala/org/apache/spark/sql/DataFrame.scala#L1173), we should not report error here.

Actually I'm wondering why we need to do checking at python side(not only this one)? Can we just call the scala API and catch the java exception?